File: leetcode/data_struct/string/0014/1.go
// auther: phoenix
// date: 2023-06-10 14:13:46
package main

import (
	"fmt"
)

/*


 */

func main() {
	longestCommonPrefix1Test()
}

func longestCommonPrefix1Test() {
	// fmt.Println(longestCommonPrefix1([]string{"flower", "flow", "flight"}))
	// fmt.Println(longestCommonPrefix1([]string{"dog", "racecar", "car"}))
	// fmt.Println(longestCommonPrefix1([]string{"cir", "car"}))
	fmt.Println(longestCommonPrefix1([]string{"ab", "a"}))
}

func longestCommonPrefix1(strs []string) string {
	length := len(strs)
	if length <= 0 {
		return ""
	}

	first := strs[0]
	firstLength := len(strs[0])
	str := ""

	for i := 0; i < firstLength; i++ {
		firstChar := first[i]
		for j := 0; j < length; j++ {
			if len(strs[j]) <= i {
				return str
			}
			if firstChar != strs[j][i] {
				return str
			}
		}
		str += string(firstChar)
	}

	return str
}

File: leetcode/data_struct/string/0014/1.ts
// auther: phoenix
// date: 2023-06-10 14:13:55

/*


*/

demo()
function demo() {
  longestCommonPrefix1Test()
}


function longestCommonPrefix1Test(){
  console.log(longestCommonPrefix1(["flower","flow","flight"]))
  console.log(longestCommonPrefix1(["dog","racecar","car"]))
  console.log(longestCommonPrefix1(["cir","car"]))

  
}

function longestCommonPrefix1(strs: string[]): string{
  const length:number = strs.length
  if (length <= 0) {
    return ''
  }

  let str:string = ''
  const first = strs[0]
  const firstLength:number = first.length
  for (let i = 0; i < firstLength; i++) {
    let firstChar = first[i];
    for (let j = 0; j < length; j++) {
      if (strs[j].length <= i) {
				return str
			}
      if (firstChar !== strs[j][i]) {
        return str
      }
    }
    str += firstChar
  }

  return str
}